src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php

```diff
@@ -66,23 +67,52 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
             "audio/x-mpg",
             "audio/x-mpegaudio",
             "audio/x-wav",
-            "audio/wav"
+            "audio/wav",
+            "audio/ogg",
+            "audio/webm",
+            "video/webm"
         );
         $allowedExtensions = array(
             ".mp3",
             ".mpa",
             ".mpg",
             ".m4a",
-            ".wav"
+            ".wav",
+            ".ogg",
+            ".webm"
         );
 
         $response = new UploadResponse();
+
         if (in_array(strtolower($fileType), $allowedTypes) && in_array(strtolower($fileExt), $allowedExtensions)) {
 
             // make the folders if they don't exist
             $project->createAssetsFolders();
             $folderPath = $project->getAudioFolderPath();
 
+            // convert audio file to mp3 or wav format if necessary //LHM
+            // FLEx only supports mp3 or wav format as of 2022-09
+
+            //First, find the duration of the file
+            $ffprobeCommand = 'ffprobe -i ' . $tmpFilePath . ' -show_entries format=duration -v quiet -of csv="p=0" 2> /dev/null';
+            $audioDuration = floatval(shell_exec($ffprobeCommand));
+
+            // Convert to .wav if the result will be less than 1 MB at 165Kb/s (recording is shorter than 6 seconds)
+            // and .mp3 otherwise (recording is longer than 6 seconds)
+            if($audioDuration < 6){
+                shell_exec('ffmpeg -i ' . $tmpFilePath . ' -b:a 165K -maxrate 165K -bufsize 80K ' . trim($tmpFilePath, '.webm') . '.wav 2> /dev/null');
+                shell_exec('mv ' . $tmpFilePath . ' ' . trim($tmpFilePath, '.webm') . '.wav 2> /dev/null');
```

Comment:
I realize this wasn't in the spec, but I can see real value in keeping the .webm file around alongside the wav and mp3 files instead of getting rid of it, even if we don't reference it.  So I would like to see lines like:
- convert webm to flex file (wav/mp3)
- move webm and flex file to assets folder

Reply:
Got it. Working on this. Currently, our uploadAudioFile() function in LexUploadCommands.php uploads one file at a time. I am working on having it upload both.

Reply:
Code has been updated to keep both the original, uploaded/recorded format, as well as the FLEx-compatible format, in case originals might be useful to have at a later date.
![image](https://user-images.githubusercontent.com/56163492/191924991-8b4e8e25-b990-4c97-a739-da8b237f0cad.png)

Reply:
Support for FLAC and OGG uploads has also been added, addressing Issue #1399

Reply:
awesome!!!